### ISSUE TITLE
In layer Filter "lower" function not working  with driacritrical marks

### ISSUE THREAD

**Issue by And121275** [qgis/QGIS#58893]
### What is the bug or the crash?

In layer Filter "lower" function not working  with driacritrical marks.
GPKG does not support ILIKE so my idea was to use lower function instated of ILIKE.
Everything is OK till text value doesn't contain driacritrical marks.
When I use layer filter: `lower("City")='żytniów'` i have no results.
When I use: `lower("City")='Żytniów' OR lower("City")='ŻytniÓw'` everything works fine
The goal is to easy filer large data with different large of letters in cities names.
Of course it is possible to convert cities name do lower letters in field calculator and then filtering will be easy.

### Steps to reproduce the issue

Make simply gpkg, utf-8 dataset with text field City and attributes: Żytniów, ŻYTNIÓW, Paris, PARIS
Set layer Filter: `lower("City")='żytniów'` and see no results
Try layer Filter: `lower("City")='paris'` and see correct two results

### Versions

Wersja QGIS
3.38.3-Grenoble
Rewizja QGIS
37f9e6efee
Wersja Qt
5.15.13
Wersja Pythona
3.12.6
Wersja GDAL/OGR
3.9.2
Wersja PROJ
9.4.0
Wersja bazy danych Rejestru EPSG
v11.004 (2024-02-24)
Wersja GEOS
3.12.2-CAPI-1.18.2
Wersja SQLite
3.45.1
Wersja PDAL
2.6.3
Wersja klienta PostgreSQL
16.2
Wersja SpatiaLite
5.1.0
Wersja QWT
6.2.0
Wersja QScintilla2
2.14.1
Wersja systemu operacyjnego
Windows 11 Version 2009

Active Python plugins
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by roya0045** [comment]
Yeah the change would probably have to go through `QLocal` instead of `QString`, Not sure if there's an easy fix when dealing with localisation.

**Comment by elpaso** [comment]
Seems to be working here:

![immagine](https://github.com/user-attachments/assets/c8742fc9-12b6-4bcf-a96e-734430f55c02)
